common/buildcraft/silicon/tile/TileLaser.java

```diff
@@ -129,11 +130,18 @@ public long getMaxPowerPerTick() {
 
     @Override
     public void update() {
+        ticks++;
         if (world.isRemote) {
+            // set laser render position on client side
+            if (ticks % (5 + world.rand.nextInt(10)) == 0 || targetPos == null) {
+                updateLaser();
+            }
             return;
         }
+        // set target tile on server side
         avgPower.tick();
-        ticks++;
+
+        BlockPos previousTargetPos = targetPos != null ? targetPos.toImmutable() : null;
```

Comment:
So now `previousTargetPos` is used only once. I think it could be inlined, an making it immutable is useless here (as you are using it only for comparing, and because this method only makes sense in `MutableBlockPos`, in normal `BlockPos`es it's just `return this;`).

Reply:
My reason for using `toImmutable` was to get a copy of the target position without pointing a new reference to the same `BlockPos` in memory. Is there a better way to get a copy of a `BlockPos`?

Reply:
BlockPos is immutable, so there's very little point in making a copy. When you change targetPos later, it creates a new, different block pos.